lms/views/api/d2l/files.py

```diff
@@ -23,15 +30,36 @@ def _find_files(modules):
 )
 def list_files(_context, request):
     """Return the list of files in the given course."""
+    course_id = request.matchdict["course_id"]
     modules = request.find_service(D2LAPIClient).course_table_of_contents(
-        org_unit=request.matchdict["course_id"]
+        org_unit=course_id
     )
     return [
         {
-            "id": f"d2l://content-resource/{file['id']}/",
+            "id": f"d2l://file/course/{course_id}/file_id/{file['id']}/",
             "display_name": file["name"],
             "updated_at": file["updated_at"],
             "type": "File",
         }
         for file in _find_files(modules)
     ]
+
+
+@view_config(
+    request_method="GET",
+    route_name="d2l_api.courses.files.via_url",
+    renderer="json",
+    permission=Permissions.API,
+)
+def via_url(_context, request):
+    course_id = request.matchdict["course_id"]
+    document_url = request.params["document_url"]
+    file_id = DOCUMENT_URL_REGEX.search(document_url)["file_id"]
+
+    public_url = request.find_service(D2LAPIClient).public_url(course_id, file_id)
+
+    access_token = request.find_service(name="oauth2_token").get().access_token
```

Comment:
I thought I might need to get this from a private method in the client but we can just use the oauth2 service :+1: